spec/furniture/marketplace/buying_products_system_spec.rb

```diff
@@ -26,16 +26,19 @@ def url_options
 
   it "Works for Guests" do # rubocop:disable RSpec/ExampleLength
     visit(polymorphic_path(marketplace.room.location))
-    set_delivery_details(delivery_address: "123 N West St Oakland, CA",
-      delivery_area: marketplace.delivery_areas.first,
-      contact_email: "AhsokaTano@example.com",
-      contact_phone_number: "1234567890")
+    select(marketplace.delivery_areas.first.label, from: "cart[delivery_area_id]")
+    click_on("Save changes")
 
     add_product_to_cart(marketplace.products.first)
 
     expect(page).to have_content("Total: #{humanized_money_with_symbol(marketplace.products.first.price + marketplace.delivery_areas.first.price)}")
 
-    click_link_or_button("Checkout")
+    click_on("Checkout")
```

Comment:
I like that this helper is more agnostic. Have we ever discussed special data attributes for test targets in UI code? This is a pattern I particular like for even further separating the markup from behavior. For example, Assuming rspec helpers allow for css selectors: 

```ruby
click_on('[data-test="checkout"]')
```

Reply:
Let's wait until we start experiencing brittleness that requires that. I'm of two minds:

1. "The UI *copy* should not be part of system tests" (which would encourage usage of structural selectors rather than textual selectors)
2. "The less attributes we *need* to add to make a system test, the better, since it's less code"

Right now I'm leaning pretty heavily towards 2; but I'm not opposed to using test ids in contexts where they are useful.

Reply:
Yeah I agree there's not enough copy/code churn that makes a strong case at this point. It's a noop at this time. 

Although, regarding your first point, in my experience structural selectors can actually be more brittle than UI copy since devs often futz with the markup structure more often in response to capability changes than the user-facing verbiage.

Reply:
Yea, I am thinking more like `#order_12345_checkout_form input[type="submit"]` than `div.foo .bar ...`